src/app/client/src/app/modules/observation/components/observation-listing/observation-listing.component.ts

```diff
@@ -0,0 +1,398 @@
+import {
+  IInteractEventEdata,
+  IImpressionEventInput,
+  TelemetryService,
+} from "@sunbird/telemetry";
+import {
+  takeUntil,
+  map,
+  delay,
+  first,
+  debounceTime,
+  tap,
+  mergeMap,
+} from "rxjs/operators";
+import { CacheService } from "ng2-cache-service";
+import { ContentManagerService } from "../../../public/module/offline/services/content-manager/content-manager.service";
+import { ObservationUtilService } from "../../service";
+import {Location} from '@angular/common';
+
+@Component({
+  selector: "app-observation-listing",
+  templateUrl: "./observation-listing.component.html",
+  styleUrls: ["./observation-listing.component.scss"],
+})
+export class ObservationListingComponent
+  implements OnInit, OnDestroy, AfterViewInit
+{
+  pageTitleSrc = "resourceService?.frmelmnts?.lbl?.observation";
+  svgToDisplay = "textbooks-banner-img.svg";
+  contentList: any = [];
+  public unsubscribe$ = new Subject<void>();
+  layoutConfiguration: any;
+  contentData;
+  contentName: string;
+  public inViewLogs = [];
+  public telemetryImpression: IImpressionEventInput;
+  public cardIntractEdata: IInteractEventEdata;
+  public showLoader = true;
+  public initFilters = false;
+  isDesktopApp = false;
+  selectedFilters: any;
+  totalCount: any = 0;
+  FIRST_PANEL_LAYOUT;
+  SECOND_PANEL_LAYOUT;
+  public allTabData;
+  config;
+  searchData: any = "";
+  public numberOfSections = new Array(
+    this.configService.appConfig.SEARCH.PAGE_LIMIT
+  );
+  public paginationDetails: IPagination;
+  queryParam: any = {};
+  showEditUserDetailsPopup:any= true;
+  payload:any;
+  constructor(
+    public searchService: SearchService,
+    public router: Router,
+    public activatedRoute: ActivatedRoute,
+    public paginationService: PaginationService,
+    public resourceService: ResourceService,
+    public toasterService: ToasterService,
+    public changeDetectorRef: ChangeDetectorRef,
+    public configService: ConfigService,
+    public utilService: UtilService,
+    public coursesService: CoursesService,
+    private playerService: PlayerService,
+    public userService: UserService,
+    public cacheService: CacheService,
+    public browserCacheTtlService: BrowserCacheTtlService,
+    public orgDetailsService: OrgDetailsService,
+    public navigationhelperService: NavigationHelperService,
+    public layoutService: LayoutService,
+    private schemaService: SchemaService,
+    public contentManagerService: ContentManagerService,
+    public telemetryService: TelemetryService,
+    private offlineCardService: OfflineCardService,
+    private kendraService: KendraService,
+    config: ConfigService,
+    private observationUtil:ObservationUtilService,
+    private location:Location,
+  ) {
+    this.config = config;
+    this.layoutConfiguration = this.layoutService.initlayoutConfig();
+    this.paginationDetails = this.paginationService.getPager(0,1,this.configService.appConfig.SEARCH.PAGE_LIMIT);
+  }
+
+  async ngOnInit(){
```

Comment:
Please make sure you're doing the cleanup on component destory like unsubsribing to any active subscriptions to avoid memory leaks

Reply:
yes we are doing it inside destroy